pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -164,12 +164,14 @@ protected ConsumerBase(PulsarClientImpl client, String topic, ConsumerConfigurat
             this.batchReceivePolicy = BatchReceivePolicy.DEFAULT_POLICY;
         }
 
+        initReceiverQueueSize();
+    }
+
+    protected void triggerBatchReceiveTimeoutTask() {
```

Comment:
If I am reading this correctly, this method will get called with every call to `batchReceive` or `batchReceiveAsync`. That means we could end up with several concurrent timer tasks for a single consumer. The current solution only has one task at a time per consumer. Given that the `doPendingBatchReceiveTask` method schedules the next task based on the `diff` to the next `opBatchReceive` timeout, I think this method should only schedule a timer task when `batchReceiveTimeout` is `null`.

Reply:
> If I am reading this correctly, this method will get called with every call to batchReceive or batchReceiveAsync.

No, we only call this method if no messages in the receiver queue. 

> I think this method should only schedule a timer task when batchReceiveTimeout is null.

Yes, make sense. I think I missed this part.

Reply:
@michaeljmarshall Fixed.

Reply:
Thanks!